File: java-systempromptleakage/vulnerable11.java
package main;

import dev.langchain4j.data.message.SystemMessage;
import dev.langchain4j.data.message.UserMessage;
import dev.langchain4j.model.chat.ChatLanguageModel;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

@RestController
public class ExportController {

    private final ChatLanguageModel model;
    private final String systemPrompt = System.getenv("SYSTEM_PROMPT");

    public ExportController(ChatLanguageModel model) {
        this.model = model;
    }

    @GetMapping("/export")
    public String exportConfig() throws IOException {
        Files.writeString(Path.of("/var/www/exports/config.txt"), "prompt=" + systemPrompt);
        return "exported";
    }

    @PostMapping("/chat")
    public String chat(@RequestParam String message) {
        return "Note: AI-generated content. Verify independently.\n\n" + model.generate(
                SystemMessage.from(systemPrompt),
                UserMessage.from(message)
        ).content().text();
    }
}

File: csharp-systempromptleakage/vulnerable11.cs
using Microsoft.AspNetCore.Mvc;
using OpenAI.Chat;

namespace ChatApi;

[ApiController]
[Route("api")]
public class ExportController : ControllerBase
{
    private readonly ChatClient _chatClient;
    private readonly string _systemPrompt = LoadPromptFromConfig();

    public ExportController(ChatClient chatClient) => _chatClient = chatClient;

    [HttpPost("export")]
    public async Task<IActionResult> ExportConfig()
    {
        await File.WriteAllTextAsync("/var/www/exports/config.txt", $"prompt={_systemPrompt}");
        return Ok(new { status = "exported" });
    }

    [HttpPost("chat")]
    public async Task<IActionResult> Chat([FromBody] ChatRequest req)
    {
        var result = await _chatClient.CompleteChatAsync(
        [
            ChatMessage.CreateSystemMessage(_systemPrompt),
            ChatMessage.CreateUserMessage(req.Message)
        ], new ChatCompletionOptions { MaxOutputTokenCount = 1024 });
        return Ok(new { reply = result.Value.Content[0].Text, disclaimer = "AI-generated content. Verify independently." });
    }

    private static string LoadPromptFromConfig() =>
        "Internal assistant. Has access to customer records and pricing data.";
}

public record ChatRequest(string Message);
